### PR TITLE
expression: fix wrong behavior in values function for Bit(1).

### LINKED ISSUE #15310: Get error data too long when use values for bit(1).
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?

```
create table t(id int primary key auto_increment, a bit(1));
insert into t values(1,1);
insert into t values(1,1) on duplicate key update id = values(id),a = values(a);
```

2. What did you expect to see?

No error.

3. What did you see instead?
```
tidb> create table t(id int primary key auto_increment, a bit(1));
Query OK, 0 rows affected (0.01 sec)

tidb> insert into t values(1,1);
Query OK, 1 row affected (0.00 sec)

tidb> insert into t values(1,1) on duplicate key update id = values(id),a = values(a);
ERROR 1406 (22001): Data Too Long, field len 1
```
4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?
master  and release-3.0.10

### PR DESCRIPTION
**Pull request by wshwsh12**
### What problem does this PR solve? 

Fixes #15310 

### What is changed and how it works?

Check the data length in built-in function ValuesInt().

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

Side effects

 - Possible performance regression
 - Increased code complexity

Related changes

 - Need to cherry-pick to the release branch 2.1 and 3.0

Release note

Fix wrong behavior in values function for type bit(n).

### CODE REVIEW

File: expression/builtin_other.go [hunk 1]
```diff
@@ -797,6 +797,19 @@ func (b *builtinValuesIntSig) evalInt(_ chunk.Row) (int64, bool, error) {
 		if row.IsNull(b.offset) {
 			return 0, true, nil
 		}
+		// For BinaryLiteral, see issue #15310
+		val := row.GetRaw(b.offset)
+		if len(val) > 8 {
+			return 0, true, errors.New("Session current insert values is too long")
+		}
+		if len(val) < 8 {
```
Comment (qw4990) [inline expression/builtin_other.go]: Regarding it as a binary number when the length is less than `8` seems tricky, but I don't have any other idea which can fix this issue easily, do you have any suggestions? @SunRunAway

Comment (SunRunAway) [reply expression/builtin_other.go]: How about checking the `FieldType` of args[0] whether is `mysql.TypeBit`?

Comment (wshwsh12) [reply expression/builtin_other.go]: I have a try, but the function doesn't have args...  len(args) == 0.  @SunRunAway

**Pull request merged**